var/spack/repos/builtin/packages/py-h5py/package.py

```diff
@@ -44,12 +50,26 @@ class PyH5py(PythonPackage):
     depends_on('mpi', when='+mpi')
     depends_on('py-mpi4py', when='+mpi', type=('build', 'run'))
 
+    # Conflicts
+    conflicts('^hdf5@1.12.0:', when='@:2.99')
```

Comment:
This will work for the new concretizer, but not for the old concretizer. Let's format this as a `depends_on`:
```python
depends_on('hdf5@1.8.4:1.11+hl', when='@:2')
depends_on('hdf5@1.8.4:+hl', when='@3:')
```

Reply:
Done.